### PR TITLE
Music: new timeline order

### PR DESCRIPTION
**Pull request by breville**
This adds a new idea for the timeline ordering, behind the `?timeline-layout-2=true` URL parameter.  

Until now, sounds have been grouped by the function that directly generates them, in adjacent rows. 

In this new ordering, all sounds generated somewhere under `when run` are ordered in rows by their play time.

### before

Screenshot 2024-05-28 at 10 45 50 PM

### after

Screenshot 2024-05-28 at 10 46 19 PM

### CODE REVIEW

File: apps/src/music/views/TimelineSimple2Events.tsx [hunk 1]
```diff
@@ -116,7 +119,32 @@ interface TimelineSimple2EventsProps {
...
   TimelineSimple2EventsProps
 > = ({paddingOffset, barWidth, eventVerticalSpace, getEventHeight}) => {
-  const soundEvents = useMusicSelector(state => state.music.playbackEvents);
+  const soundEventsOriginal = useMusicSelector(
+    state => state.music.playbackEvents
+  );
+
+  const soundEvents = useTimelineLayout2
+    ? [...soundEventsOriginal].sort((a, b) => {
+        if (a.triggered && b.triggered) {
```
Comment (fisher-alice) [inline apps/src/music/views/TimelineSimple2Events.tsx]: Nit: You could pull the compare function out and name it something like `sortByPlaytime`.

File: apps/src/music/views/TimelineSimple2Events.tsx [hunk 2]
```diff
@@ -116,7 +119,32 @@ interface TimelineSimple2EventsProps {
 const TimelineSimple2Events: React.FunctionComponent<
   TimelineSimple2EventsProps
 > = ({paddingOffset, barWidth, eventVerticalSpace, getEventHeight}) => {
-  const soundEvents = useMusicSelector(state => state.music.playbackEvents);
+  const soundEventsOriginal = useMusicSelector(
```
Comment (fisher-alice) [inline apps/src/music/views/TimelineSimple2Events.tsx:151]: You could add comment that includes how these events are sorted as opposed to `useTimelineLayout2`.

**Review by fisher-alice — APPROVED** [review]
Cool investigation. I pulled locally and appreciated the timeline events in play time order.

File: apps/src/music/views/TimelineSimple2Events.tsx (hunk 1 above)
Comment (breville) [reply apps/src/music/views/TimelineSimple2Events.tsx]: Good call.  Extracted the whole conversion into a new helper function.

File: apps/src/music/views/TimelineSimple2Events.tsx (hunk 2 above)
Comment (breville) [reply apps/src/music/views/TimelineSimple2Events.tsx:151]: Good call.  Added more comments.

**Pull request merged**